collector/lib/CollectorConfig.cpp

```diff
@@ -447,6 +451,73 @@ void CollectorConfig::HandleConfig(const std::filesystem::path& filePath) {
   }
 }
 
+void WaitForFileToExist(const std::filesystem::path& filePath) {
+  while (!std::filesystem::exists(filePath)) {
+    sleep(5);
```

Comment:
Is this stopping the entirety of collector or just the inotify thread?

Reply:
Also, have you tested how long k8s takes to recreate the config file when updated? 5 seconds sounds like a lot, we can probably check every second.
```suggestion
    sleep(1);
```

Reply:
It just stops the inotify thread.